src/compiler/checker.ts

```diff
@@ -17843,6 +17846,46 @@ namespace ts {
             const symbol = getGlobalSymbol("ReadonlyArray", SymbolFlags.Type, /*diagnostic*/ undefined);
             globalReadonlyArrayType = symbol && <GenericType>getTypeOfGlobalSymbol(symbol, /*arity*/ 1);
             anyReadonlyArrayType = globalReadonlyArrayType ? createTypeFromGenericGlobalType(globalReadonlyArrayType, [anyType]) : anyArrayType;
+
+            // If we have specified that we are importing helpers, we should report global
+            // errors if we cannot resolve the helpers external module, or if it does not have
+            // the necessary helpers exported.
+            if (compilerOptions.importHelpers) {
+                forEach(host.getSourceFiles(), sourceFile => {
+                    const requestedHelpers = sourceFile.flags & NodeFlags.EmitHelperFlags;
+                    if (requestedHelpers && (compilerOptions.isolatedModules || isExternalModule(sourceFile))) {
+                        const helpers = resolveExternalModule(sourceFile, externalHelpersModuleNameText, Diagnostics.Cannot_find_module_0, /*errorNode*/ undefined);
+                        if (helpers) {
+                            const exports = helpers.exports;
+                            if (requestedHelpers & NodeFlags.HasClassExtends && languageVersion < ScriptTarget.ES6) {
+                                verifyHelperSymbol(exports, "__extends", SymbolFlags.Value);

```

Comment:
consider returning after the first error, to avoid reporting the same error for every file.